### PR TITLE
Bump Go version to 1.21.6

### PR DESCRIPTION
**Pull request by rdavid**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -3,7 +3,7 @@ BINARY_NAME=gradetool
 REGISTRY?=quay.io
 TNF_IMAGE_NAME?=testnetworkfunction/gradetool
 IMAGE_TAG?=latest
-GOLANGCI_VERSION=v1.55.1
+GOLANGCI_VERSION=v1.55.2
```
Comment (sebrandon1) [inline Makefile:6]: This should be a separate PR correct?

**Review by sebrandon1 — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (rdavid) [reply Makefile:6]: The linter fails when not updated [0].

[0] https://github.com/golangci/golangci-lint/issues/3718
